### ISSUE TITLE
GET /leases - returns 404 when there are no records matching query parameters

### ISSUE THREAD

**Issue by AmudaPalani** [Optum/dce#204]
**Version information**

This was found in PR https://github.com/Optum/dce/pull/205/files

**Describe the bug**

When there are no records satisfying query parameters, GET /leases returns 404. It should return an empty array.
**To Reproduce**

1. Go to API Gateway
2. Select /leases GET endpoint
3. Ensure no records exist in Leases table
4. Run the above endpoint with query parameter status=Active

Expected output:
Returns 200 OK with an empty array 

Actual output:
Returns a 404 error
**Expected behavior**

**Additional context**

**Comment by eschwartz** [comment]
Eg `GET /leases?status=Active&principalId=me@example.com` returns a 404, if I don't have an active lease.

This propagates to the CLI:

```
$ dce leases list -s Active -p me@example.com
> unknown error (status 404)
$ echo $?
> 1 
```

**Comment by kddejong** [comment]
Fixed in #205
